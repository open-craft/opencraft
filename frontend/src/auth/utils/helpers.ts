import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
