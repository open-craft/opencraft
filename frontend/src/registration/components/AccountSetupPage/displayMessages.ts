const messages = {
  createYourAccount: {
    defaultMessage: 'Create your account',
    description: ''
  },
  fullName: {
    defaultMessage: 'Your full name*',
    description: ''
  },
  fullNameHelp: {
    defaultMessage: 'First and last name.',
    description: ''
  },
  username: {
    defaultMessage: 'Username*',
    description: ''
  },
  usernameHelp: {
    defaultMessage:
      'The username for all the services you will use to create your courses.',
    description: ''
  },
  email: {
    defaultMessage: 'Email address*',
    description: ''
  },
  emailHelp: {
    defaultMessage:
      'This is also your account name, and where we will send important notices.',
    description: ''
  },
  password: {
    defaultMessage: 'Password*',
    description: ''
  },
  passwordHelp: {
    defaultMessage:
      'Password must have 8 characters, 1 number, 1 capital letter and 1 special character.',
    description: ''
  },
  passwordConfirm: {
    defaultMessage: 'Confirm password*',
    description: ''
  },
  passwordConfirmHelp: {
    defaultMessage: 'Confirm the password in the field above.',
    description: ''
  },
  acceptTos: {
    defaultMessage:
      'I accept that this is a free trial, and that the Open edX dedicated site is ' +
      'provided without any guarantee. {tos} and {privacy_policy}.*',
    description: ''
  },
  acceptDomainCondition: {
    defaultMessage:
      'By submitting the registration form, I assert that I have ' +
      'rights to use the domain I provided.*',
    description: ''
  },
  subscribeToUpdates: {
    defaultMessage:
      'I want OpenCraft to occasionally email me about important news, tips and new features.',
    description: ''
  },
  privacyPolicy: {
    defaultMessage: 'Privacy Policy',
    description: ''
  },
  termsOfService: {
    defaultMessage: 'Terms of Service',
    description: ''
  }
};

export default messages;
