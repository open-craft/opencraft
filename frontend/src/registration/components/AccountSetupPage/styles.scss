@import '~styles/theme';

.account-form {
  width: 100%;

  h2 {
    font-family: PlayfairDisplay;
    font-size: 30px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: 1.8px;
    color: #1e4c59;
  }

  a {
    text-decoration: underline;
  }
}

.invalid-feedback-checkbox {
  font-family: $chivo-font;
  font-size: 12px;
  letter-spacing: 0.72px;
  color: red;
}
