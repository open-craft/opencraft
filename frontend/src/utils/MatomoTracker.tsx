import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
