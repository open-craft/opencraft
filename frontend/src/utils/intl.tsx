import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
