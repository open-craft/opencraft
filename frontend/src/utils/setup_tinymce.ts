import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
