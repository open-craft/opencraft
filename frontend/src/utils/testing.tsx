import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
