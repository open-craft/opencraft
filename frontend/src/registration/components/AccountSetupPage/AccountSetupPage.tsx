import { RootState } from 'global/state';
import * as React from 'react';
import { connect } from 'react-redux';
import { Form } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import {
  RegistrationNavButtons,
  RedirectToCorrectStep
} from 'registration/components';
import { TextInputField } from 'ui/components';
import {
  RegistrationSteps,
  PRIVACY_POLICY_LINK,
  TOS_LINK
} from 'global/constants';
import { RegistrationStateModel } from 'registration/models';
import { clearErrorMessage, submitRegistration } from '../../actions';
import { RegistrationPage } from '../RegistrationPage';
import messages from './displayMessages';
import './styles.scss';

interface ActionProps {
  submitRegistration: Function;
  clearErrorMessage: Function;
}

interface State {
  [key: string]: string | boolean;
  fullName: string;
  username: string;
  email: string;
  password: string;
  passwordConfirm: string;
  acceptTOS: boolean;
  acceptPaidSupport: boolean;
  acceptDomainCondition: boolean;
  subscribeToUpdates: boolean;
}

interface StateProps extends RegistrationStateModel {}
interface Props extends StateProps, ActionProps {
  history: {
    goBack: Function;
  };
}

@connect<StateProps, ActionProps, {}, Props, RootState>(
  (state: RootState) => ({
    ...state.registration
  }),
  {
    submitRegistration,
    clearErrorMessage
  }
)
export class AccountSetupPage extends React.PureComponent<Props, State> {
  constructor(props: Props, state: State) {
    super(props);

    this.state = {
      fullName: this.props.registrationData.fullName,
      username: this.props.registrationData.username,
      email: this.props.registrationData.email,
      password: this.props.registrationData.password,
      passwordConfirm: this.props.registrationData.passwordConfirm,
      acceptTOS: this.props.registrationData.acceptTOS,
      acceptPaidSupport: this.props.registrationData.acceptPaidSupport,
      acceptDomainCondition: this.props.registrationData.acceptDomainCondition,
      subscribeToUpdates: this.props.registrationData.subscribeToUpdates
    };
  }

  private onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const field = e.target.name;
    let value: boolean | string;

    if (e.target.type === 'checkbox') {
      value = e.target.checked;
    } else {
      value = e.target.value;
    }

    this.setState({
      [field]: value
    });

    // Clear error message when the user changes field
    if (this.props.registrationFeedback[field]) {
      this.props.clearErrorMessage(field);
    }
  };

  private submitRegistration = () => {
    this.props.submitRegistration(
      {
        fullName: this.state.fullName,
        username: this.state.username,
        email: this.state.email,
        password: this.state.password,
        passwordConfirm: this.state.passwordConfirm,
        acceptTOS: this.state.acceptTOS,
        acceptPaidSupport: this.state.acceptPaidSupport,
        acceptDomainCondition: this.state.acceptDomainCondition,
        subscribeToUpdates: this.state.subscribeToUpdates
      },
      {
        externalDomain: this.props.registrationData.externalDomain,
        subdomain: this.props.registrationData.subdomain,
        instanceName: this.props.registrationData.instanceName
      },
      RegistrationSteps.CONGRATS
    );
  };

  render() {
    const checkboxLinks = {
      tos: (
        <a href={PRIVACY_POLICY_LINK} target="_blank" rel="noopener noreferrer">
          <WrappedMessage id="privacyPolicy" messages={messages} />
        </a>
      ),
      privacy_policy: (
        <a href={TOS_LINK} target="_blank" rel="noopener noreferrer">
          <WrappedMessage id="termsOfService" messages={messages} />
        </a>
      )
    };
    return (
      <RegistrationPage
        title="Create your Pro & Teacher Account"
        currentStep={3}
      >
        <RedirectToCorrectStep
          currentPageStep={3}
          currentRegistrationStep={this.props.currentRegistrationStep}
        />
        <Form className="account-form">
          <h2>
            <WrappedMessage id="createYourAccount" messages={messages} />
          </h2>
          <TextInputField
            fieldName="fullName"
            value={this.state.fullName}
            onChange={this.onChange}
            messages={messages}
            error={this.props.registrationFeedback.fullName}
          />
          <TextInputField
            fieldName="username"
            value={this.state.username}
            onChange={this.onChange}
            messages={messages}
            error={this.props.registrationFeedback.username}
          />
          <TextInputField
            fieldName="email"
            value={this.state.email}
            onChange={this.onChange}
            type="email"
            messages={messages}
            error={this.props.registrationFeedback.email}
          />
          <TextInputField
            fieldName="password"
            value={this.state.password}
            onChange={this.onChange}
            type="password"
            messages={messages}
            error={this.props.registrationFeedback.password}
          />
          <TextInputField
            fieldName="passwordConfirm"
            value={this.state.passwordConfirm}
            onChange={this.onChange}
            type="password"
            messages={messages}
            error={this.props.registrationFeedback.passwordConfirm}
          />
          <Form.Check type="checkbox" id="acceptTOS" custom>
            <Form.Check.Input
              type="checkbox"
              name="acceptTOS"
              checked={this.state.acceptTOS}
              onChange={this.onChange}
            />
            <Form.Check.Label>
              <WrappedMessage
                id="acceptTos"
                messages={messages}
                values={checkboxLinks}
              />
            </Form.Check.Label>
          </Form.Check>
          {this.props.registrationFeedback.acceptTOS && (
            <div className="invalid-feedback-checkbox">
              {this.props.registrationFeedback.acceptTOS}
            </div>
          )}
          <Form.Check type="checkbox" id="acceptPaidSupport" custom>
            <Form.Check.Input
              type="checkbox"
              name="acceptPaidSupport"
              checked={this.state.acceptPaidSupport}
              onChange={this.onChange}
            />
            <Form.Check.Label>
              <WrappedMessage id="acceptPaidSupport" messages={messages} />
            </Form.Check.Label>
          </Form.Check>
          {this.props.registrationFeedback.acceptPaidSupport && (
            <div className="invalid-feedback-checkbox">
              {this.props.registrationFeedback.acceptPaidSupport}
            </div>
          )}
          <Form.Check type="checkbox" id="acceptDomainCondition" custom>
            <Form.Check.Input
              type="checkbox"
              name="acceptDomainCondition"
              checked={this.state.acceptDomainCondition}
              onChange={this.onChange}
            />
            <Form.Check.Label>
              <WrappedMessage id="acceptDomainCondition" messages={messages} />
            </Form.Check.Label>
          </Form.Check>
          {this.props.registrationFeedback.acceptDomainCondition && (
            <div className="invalid-feedback-checkbox">
              {this.props.registrationFeedback.acceptDomainCondition}
            </div>
          )}
          <Form.Check type="checkbox" id="subscribeToUpdates" custom>
            <Form.Check.Input
              type="checkbox"
              name="subscribeToUpdates"
              checked={this.state.subscribeToUpdates}
              onChange={this.onChange}
            />
            <Form.Check.Label>
              <WrappedMessage id="subscribeToUpdates" messages={messages} />
            </Form.Check.Label>
          </Form.Check>
          {this.props.registrationFeedback.subscribeToUpdates && (
            <div className="invalid-feedback-checkbox">
              {this.props.registrationFeedback.subscribeToUpdates}
            </div>
          )}
        </Form>
        <RegistrationNavButtons
          loading={this.props.loading}
          disableNextButton={false}
          showBackButton
          showNextButton
          handleBackClick={() => {
            this.props.history.goBack();
          }}
          handleNextClick={this.submitRegistration}
        />
      </RegistrationPage>
    );
  }
}
