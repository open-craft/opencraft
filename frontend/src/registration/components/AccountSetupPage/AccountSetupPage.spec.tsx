import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { AccountSetupPage } from './AccountSetupPage';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<AccountSetupPage />).toJSON();
    expect(tree).toMatchSnapshot();
});
