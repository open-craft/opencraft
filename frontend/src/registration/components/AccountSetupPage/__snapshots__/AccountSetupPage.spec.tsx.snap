// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="registration-page"
>
  <h1>
    Create your Pro & Teacher Account
  </h1>
  <div
    className="step-bar-container"
  >
    <div
      className="d-flex justify-content-center"
    >
      <div
        className="step-bar"
      >
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
        <svg
          className="circled-number"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            className="outer-circle"
            cx={10}
            cy={10}
            r={10}
          />
          <circle
            className="inner-circle"
            cx={10}
            cy={10}
            r={7}
            strokeWidth={2}
          />
          <text
            className="circle-text"
            color="#1e4c59"
            dy="0.4em"
            fontSize="10px"
            strokeWidth="1px"
            textAnchor="middle"
            x="50%"
            y="50%"
          >
            3
          </text>
        </svg>
        <span
          className="line"
        />
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
      </div>
    </div>
  </div>
  <div
    className="registration-page-container"
  >
    <div
      className="registration-page-content row"
    >
      <form
        className="account-form"
      >
        <h2>
          Create your account
        </h2>
        <div
          className="text-input-container"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Your full name*
            </label>
            <input
              className="form-control"
              name="fullName"
              onChange={[Function]}
              value=""
            />
            <p>
              Name and surname.
            </p>
          </div>
        </div>
        <div
          className="text-input-container"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Username*
            </label>
            <input
              className="form-control"
              name="username"
              onChange={[Function]}
              value=""
            />
            <p>
              This would also be the username of the administrator account on the Open edX instance.
            </p>
          </div>
        </div>
        <div
          className="text-input-container"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Email address*
            </label>
            <input
              className="form-control"
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
            <p>
              This is also your account name, and where we will send important notices.
            </p>
          </div>
        </div>
        <div
          className="text-input-container"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Password*
            </label>
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
            <p>
              Password must have 8 characters, 1 number, 1 capital letter and 1 special character.
            </p>
          </div>
        </div>
        <div
          className="text-input-container"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Confirm password*
            </label>
            <input
              className="form-control"
              name="passwordConfirm"
              onChange={[Function]}
              type="password"
              value=""
            />
            <p>
              Confirm the password in the field above.
            </p>
          </div>
        </div>
        <div
          className="custom-control custom-checkbox"
        >
          <input
            checked={false}
            className="custom-control-input"
            id="acceptTOS"
            name="acceptTOS"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="acceptTOS"
          >
            I accept that this is a free trial, and that the instance is provided without any guarantee. 
            <a
              href="/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              Privacy Policy
            </a>
             and 
            <a
              href="/#"
              rel="noopener noreferrer"
              target="_blank"
            >
              Terms of Service
            </a>
            .*
          </label>
        </div>
        <div
          className="custom-control custom-checkbox"
        >
          <input
            checked={false}
            className="custom-control-input"
            id="acceptPaidSupport"
            name="acceptPaidSupport"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="acceptPaidSupport"
          >
            I understand that I can email OpenCraft with questions and that requests for support are subject to hourly fees.*
          </label>
        </div>
        <div
          className="custom-control custom-checkbox"
        >
          <input
            checked={false}
            className="custom-control-input"
            id="acceptDomainCondition"
            name="acceptDomainCondition"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="acceptDomainCondition"
          >
            By submitting the registration form, I assert that I have rights to use the domain I provided.*
          </label>
        </div>
        <div
          className="custom-control custom-checkbox"
        >
          <input
            checked={false}
            className="custom-control-input"
            id="subscribeToUpdates"
            name="subscribeToUpdates"
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="custom-control-label"
            htmlFor="subscribeToUpdates"
          >
            I want OpenCraft to occasionally email me about important news, tips and new features.
          </label>
        </div>
      </form>
      <div
        className="registration-nav"
      >
        <button
          className="float-left btn btn-outline-primary btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Back
        </button>
        <button
          className="float-right loading btn btn-primary btn-lg"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
